src/gui/transferlistmodel.cpp

```diff
@@ -712,6 +683,72 @@ QColor getColorByState(const BitTorrent::TorrentState state)
     }
 }
 
+QColor getColorByState(const BitTorrent::TorrentState state)
+{
+    switch (state) {
+    case BitTorrent::TorrentState::Downloading:
+        return UIThemeManager::instance()->getColor(QStringLiteral("TransferList.Downloading")
```

Comment:
Seems not so optimal. I would initially load all colors into hash (mapped by `BitTorrent::TorrentState`) and then do the following:
```c++
QColor getColorByState(const BitTorrent::TorrentState state)
{
    return m_stateColors.value(state, getDefaultColorByState(state));
}
```

Reply:
for now, we can't ask UIThemeManager if a given id is available, 
so I have to do it something like -
```
m_stateColor{BitTorrent::TorrentState::Downloading, UIThemeManager::instance()->getColor(QStringLiteral("TransferList.Downloading")
                                                    , getDefaultColorByState<BitTorrent::TorrentState::Downloading>())
```

then 
```
    return m_stateColors.value(state, getDefaultColorByState(state));
```
will not ever use `getDefaultColorByState` and https://github.com/qbittorrent/qBittorrent/issues/12447 will not be fixed

one way I can think of is to call UIThemeManager::getColor with an invalid color as the default value and check against it before inserting in the hash, but doesn't that will be too much

but
```
UIThemeManager::instance()->getColor(QStringLiteral("TransferList.Downloading")
                                                    , getDefaultColorByState<BitTorrent::TorrentState::Downloading>());
```
basically does the same. search in hash, if not present return default value

Reply:
the main cause of #12447 was that the `isDark` constant can change at realtime thus we shouldn't cache `getDefaultColorByState` values

Reply:
> we can't ask UIThemeManager if a given id is available,

I can introduce UIThemeManager::isColorAvailable, but wouldn't that expose details*

*I'm asking here

Reply:
>the main cause of #12447 was that the `isDark` constant can change at realtime thus we shouldn't cache `getDefaultColorByState` values to fix it

Did I suppose you to cache it? I supposed to cache the colors that come from theme.
You misunderstand my offer, and then you try to convince me that it is incorrect.

You must load the colors so that the resulting hash does contain only those that actually exist in the theme (it's a little more complicated):
```c++
// for each "ID, state" pair
const QColor themeColor = UIThemeManager::instance()->getColor(id);
if (themeColor.isValid())
    m_stateColors[state] = themeColor;
```
Then suggested code (`return m_stateColors.value(state, getDefaultColorByState(state));`) will return color from theme (if theme is used) or default (either for dark or light mode, if theme isn't used or if the theme doesn't provide an appropriate color for ID).

Reply:
>  you try to convince me that it is incorrect.

I had no such intention, I just wrote what my understanding was, Sorry if you understood otherwise.